internal/bindings/server.go

```diff
@@ -135,7 +140,8 @@ func (s *Node) SetDialFunc(dial protocol.DialFunc) error {
 	server := (*C.dqlite_node)(unsafe.Pointer(s))
 	connectLock.Lock()
 	defer connectLock.Unlock()
-	connectIndex++
+
+	// The index should have been incremented by NewNode.
```

Comment:
This will not work, what happens when someone does? (pseudocode)

```
n1 := NewNode()
n2 := NewNode()
n1.SetDialFunc(func)
n2.SetDialFunc(func)
```

Reply:
Wishful thinking :)

Changed it to just assign the context to the `Node` struct now.